media-libs/opencv/opencv-4.4.0.ebuild

```diff
@@ -0,0 +1,541 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8,9} )
+CMAKE_ECLASS=cmake
+inherit java-pkg-opt-2 java-ant-2 cmake-multilib python-r1 toolchain-funcs
+
+DESCRIPTION="A collection of algorithms and sample code for various computer vision problems"
+HOMEPAGE="https://opencv.org"
+TINY_DNN_PV="1.0.0a3"
+SRC_URI="https://github.com/${PN}/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz
+	dnnsamples? ( https://dev.gentoo.org/~amynka/snap/${PN}-3.4.0-res10_300x300-caffeemodel.tar.gz )
+	contrib? (
+		https://github.com/${PN}/${PN}_contrib/archive/${PV}.tar.gz -> ${P}_contrib.tar.gz
+		contribdnn? ( https://dev.gentoo.org/~amynka/snap/${PN}-3.4.0-face_landmark_model.tar.gz )
+		contribxfeatures2d? ( https://dev.gentoo.org/~amynka/snap/vgg_boostdesc-3.2.0.tar.gz )
+	)"
+
+LICENSE="BSD"
+SLOT="0/${PV}" # subslot = libopencv* soname version
+KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~x86"
+IUSE="contrib contribcvv contribdnn contribfreetype contribhdf contribovis contribsfm contribxfeatures2d cuda debug dnnsamples -download +eigen examples +features2d ffmpeg gdal gflags glog gphoto2 gstreamer gtk3 ieee1394 jpeg jpeg2k lapack lto opencl openexr opengl openmp opencvapps pch png +python qt5 tesseract testprograms threads tiff vaapi v4l vtk webp xine"
+
+# The following lines are shamelessly stolen from ffmpeg-9999.ebuild with modifications
+ARM_CPU_FEATURES=(
+	cpu_flags_arm_neon:NEON
+	cpu_flags_arm_vfpv3:VFPV3
+)
+PPC_CPU_FEATURES=(
+	cpu_flags_ppc_vsx:VSX
+	cpu_flags_ppc_vsx3:VSX3
+)
+X86_CPU_FEATURES_RAW=(
+	avx:AVX
+	avx2:AVX2
+	avx512f:AVX_512F
+	f16c:FP16
+	fma3:FMA3
+	popcnt:POPCNT
+	sse:SSE
+	sse2:SSE2
+	sse3:SSE3
+	ssse3:SSSE3
+	sse4_1:SSE4_1
+	sse4_2:SSE4_2
+)
+X86_CPU_FEATURES=( ${X86_CPU_FEATURES_RAW[@]/#/cpu_flags_x86_} )
+CPU_FEATURES_MAP=(
+	${ARM_CPU_FEATURES[@]}
+	${PPC_CPU_FEATURES[@]}
+	${X86_CPU_FEATURES[@]}
+)
+IUSE="${IUSE} ${CPU_FEATURES_MAP[@]%:*}"
+
+# OpenGL needs gtk or Qt installed to activate, otherwise build system
+# will silently disable it Wwithout the user knowing, which defeats the
+# purpose of the opengl use flag.
+REQUIRED_USE="
```

Comment:
Maybe you'll also add `eigen? ( !pch )` (to reflect the problem mentioned in conversation?)

Reply:
I don't think this restriction is required, AFAIK this pch issue only affects eigen-3.3.7-r1.

Reply:
Looking in the ebuilds sources I bet new versions of eigen3 will be installed the very same way as `3.3.7-r1`.
So, they will also be triggered by that.

Although, looks like there is another (and better) fix: https://github.com/opencv/opencv/issues/14868#issuecomment-664041088
We can just pass the path to eigen include dir to the opencv's cmake.

Reply:
@msva Thanks a lot for that hint! Passing the flag also fixes the build for media-libs/opencv-4.1.2-r3 (https://github.com/devurandom/gentoo-overlay/commit/9cab6197db89b165e5ce5995a397d1c8156904c2).

Reply:
devurandom/gentoo-overlay@196633deb29b999b6e07f0cc37af3929340a6ab9 fixes this for media-libs/opencv-4.4.0. It seems it was forgotten before.

See-also: https://bugs.gentoo.org/733796

Reply:
ping @rossbridger @Amynka  ^^^
how do you think?

Reply:
This bug is resolved already